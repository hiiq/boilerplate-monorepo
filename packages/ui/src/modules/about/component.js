import React from 'react';
import { Body, Content, Footer, Header, Breadcrumbs } from 'shared/Content';
import { useTranslate } from 'shared/useTranslate';

const About = () => {
  const t = useTranslate({
    component: 'about',
    namespace: 'about',
  });

  return (
    <Content>
      <Breadcrumbs />
      <Header title={t('title')} />
      <Body>{t('body')}</Body>
      <Footer>{t('footer')}</Footer>
    </Content>
  );
};

export { About };
