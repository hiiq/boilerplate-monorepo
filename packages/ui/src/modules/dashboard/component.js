import React from 'react';
import { Body, Content, Footer, Header, Breadcrumbs } from 'shared/Content';
import { useTranslate } from 'shared/useTranslate';

const Dashboard = () => {
  const t = useTranslate({
    component: 'dashboard',
    namespace: 'dashboard',
  });

  return (
    <Content>
      <Breadcrumbs />
      <Header title={t('title')} />
      <Body>
        <div
          dangerouslySetInnerHTML={{
            __html: t('body'),
          }}
        />
      </Body>
      <Footer>{t('footer')}</Footer>
    </Content>
  );
};

export { Dashboard };
