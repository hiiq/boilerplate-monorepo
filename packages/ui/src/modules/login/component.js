import { useMutation } from '@apollo/client';
import React, { useState } from 'react';
import { Button as SharedButton, Context } from 'shared/Button';
import { Body, Breadcrumb, Breadcrumbs, Content, Header } from 'shared/Content';
import { Link } from 'shared/Link';
import { Page } from 'shared/Page';
import { Mutation } from 'shared/graphql/mutation';
import { Query } from 'shared/graphql/query';
import { useAuth } from 'shared/useAuth';
import { useTranslate } from 'shared/useTranslate';
import styled from 'styled-components/macro';
import { Route } from 'types/route';

const { PRIMARY } = Context;

const Button = styled(SharedButton)`
  width: fit-content;
`;

// eslint-disable-next-line max-statements
const Login = () => {
  const t = useTranslate({
    component: 'login',
    namespace: 'login',
  });

  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const { isAuthenticated, logout, updateAccessToken } = useAuth();

  const [mutate, { loading: isDisabled }] = useMutation(Mutation.LOGIN, {
    refetchQueries: [{ query: Query.ME }],
  });

  const onLogin = async () => {
    const input = {
      password,
      username,
    };

    await mutate({
      update: (_cache, { data }) => {
        const { userLogin: accessToken } = data;

        if (!accessToken) {
          logout();

          return;
        }

        updateAccessToken(accessToken);
      },
      variables: { input },
    });
  };

  const onUsernameChange = e => {
    setUsername(e.target.value);
  };

  const onPasswordChange = e => {
    setPassword(e.target.value);
  };

  const onLogout = logout;
  const logInOutKey = isAuthenticated ? t('logout') : t('title');
  const authFn = isAuthenticated ? onLogout : onLogin;

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.SIGNUP.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body>
          {isAuthenticated ? (
            <Link to={Route.LOGOUT.path}>{t('logout')}</Link>
          ) : (
            <>
              <form onSubmit={e => e.preventDefault()}>
                <div>
                  <input
                    name="username"
                    onChange={onUsernameChange}
                    placeholder="username"
                    type="text"
                    value={username}
                  />
                </div>
                <div>
                  <input
                    name="password"
                    onChange={onPasswordChange}
                    placeholder="password"
                    type="password"
                    value={password}
                  />
                </div>
                <Button
                  context={PRIMARY}
                  disabled={isDisabled}
                  onClick={authFn}
                  text={logInOutKey}
                  type="submit"
                />
              </form>
              <br />
              or <Link to={Route.SIGNUP.path}>{t('signup')}</Link>
            </>
          )}
        </Body>
      </Content>
    </Page>
  );
};

export { Login };
