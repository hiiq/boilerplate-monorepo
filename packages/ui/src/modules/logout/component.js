import { useHistory } from 'react-router-dom';
import { useAuth } from 'shared/useAuth';
import { useTimeout } from 'shared/useTimeout';
import { Route } from 'types/route';

const Logout = () => {
  const history = useHistory();
  const { logout } = useAuth();

  useTimeout(() => history.push(Route.LOGIN.path), 1);

  logout();

  return null;
};

export { Logout };
