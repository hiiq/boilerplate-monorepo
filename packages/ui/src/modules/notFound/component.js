import { seconds } from '@puttingpoker/common';
import * as Sentry from '@sentry/browser';
import LogRocket from 'logrocket';
import { prop } from 'ramda';
import React, { useEffect, useState } from 'react';
import { useHistory, useLocation } from 'react-router-dom';
import { Body, Breadcrumb, Breadcrumbs, Content, Header } from 'shared/Content';
import { Page } from 'shared/Page';
import { useTranslate } from 'shared/useTranslate';
import styled from 'styled-components/macro';
import { LogLevel } from 'types/logLevel';
import { Route } from 'types/route';

const SECONDS_TO_REDIRECT = 15;

const logNotFound = () => {
  Sentry.withScope(scope => {
    scope.setExtra('sessionUrl', LogRocket.sessionURL);
    scope.setExtra('pathname', prop('pathname', window.location));
    scope.setTag('search', prop('search', window.location));
    scope.setLevel(LogLevel.INFO);
    Sentry.captureException(
      new Error('User visited a resource that could not be found')
    );
  });
};

const NotFound = () => {
  const t = useTranslate({
    component: 'notFound',
    namespace: 'notFound',
  });
  const [tics, setTics] = useState(SECONDS_TO_REDIRECT);
  const history = useHistory();
  const location = useLocation();

  useEffect(() => {
    logNotFound();
  }, []);

  useEffect(() => {
    let interval = null;

    if (tics <= 1) return history.goBack();

    const ticToc = () => setTics(tics - 1);

    interval = setInterval(ticToc, seconds(1));

    return () => clearInterval(interval);
  }, [history, tics, setTics]);

  const { pathname } = location;

  const Italicize = styled.span`
    font-style: italic;
    margin-left: 0.25rem;
  `;

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.NOT_FOUND.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body>
          <p>
            {t('resourceNotFound')}:<Italicize>{pathname}</Italicize>
          </p>
          <p>{t('returning', { seconds: tics })}</p>
        </Body>
      </Content>
    </Page>
  );
};

export { NotFound };
