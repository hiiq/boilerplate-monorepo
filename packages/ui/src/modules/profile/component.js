import React from 'react';
import { Body, Content, Header, Breadcrumbs, Breadcrumb } from 'shared/Content';
import { Page } from 'shared/Page';
import { useTranslate } from 'shared/useTranslate';
import { Route } from 'types/route';

const Profile = () => {
  const t = useTranslate({
    component: 'profile',
    namespace: 'profile',
  });

  return (
    <Page>
      <Content>
        <Breadcrumbs>
          <Breadcrumb isEnd text={t('title')} to={Route.USER_PROFILE.path} />
        </Breadcrumbs>
        <Header title={t('title')} />
        <Body>&nbsp;</Body>
      </Content>
    </Page>
  );
};

export { Profile };
