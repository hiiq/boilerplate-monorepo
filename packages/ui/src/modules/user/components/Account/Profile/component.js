import { always, equals } from 'ramda';
import React from 'react';
import { useLocation } from 'react-router-dom';
import { Body, Breadcrumb, Breadcrumbs, Header } from 'shared/Content';
import { useTranslate } from 'shared/useTranslate';
import { Route } from 'types/route';
import { Form } from './Form';

const Profile = () => {
  const t = useTranslate({
    component: 'user',
    namespace: 'user',
  });

  const location = useLocation();
  const isActive = equals(location.pathname, Route.USER_ACCOUNT.path);

  return (
    <>
      <Breadcrumbs>
        <Breadcrumb
          isActive={always(isActive)}
          text={t('account')}
          to={Route.USER_ACCOUNT.path}
        />
        <Breadcrumb
          isEnd
          text={t('profile')}
          to={Route.USER_ACCOUNT_PROFILE.path}
        />
      </Breadcrumbs>
      <Header title={t('profile')} />
      <Body>
        <Form />
      </Body>
    </>
  );
};

export { Profile };
