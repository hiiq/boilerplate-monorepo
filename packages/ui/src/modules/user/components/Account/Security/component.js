import { always, equals } from 'ramda';
import React from 'react';
import { useLocation } from 'react-router-dom';
import { Body, Breadcrumb, Breadcrumbs, Header } from 'shared/Content';
import { useTranslate } from 'shared/useTranslate';
import { Route } from 'types/route';
import { Form } from './Form';

const Security = () => {
  const t = useTranslate({
    component: 'user',
    namespace: 'user',
  });

  const location = useLocation();
  const isActive = equals(location.pathname, Route.USER_ACCOUNT.path);

  return (
    <>
      <Breadcrumbs>
        <Breadcrumb
          isActive={always(isActive)}
          text={t('account')}
          to={Route.USER_ACCOUNT.path}
        />
        <Breadcrumb
          isEnd
          text={t('security')}
          to={Route.USER_ACCOUNT_SECURITY.path}
        />
      </Breadcrumbs>
      <Header title={t('security')} />
      <Body>
        <Form />
      </Body>
    </>
  );
};

export { Security };
