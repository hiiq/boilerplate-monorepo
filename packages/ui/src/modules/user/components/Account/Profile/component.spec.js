import React from 'react';
import * as MutationHooks from 'shared/graphql/mutation/useUserRegister';
import { render } from 'testHelpers';
import { Profile } from './component';

jest.mock('shared/Content');
jest.mock('shared/Page');
jest.mock('shared/useAuth');

jest.mock('./Form', () => ({
  Form: props => <x-Form {...props} />,
}));

describe('<Profile/>', () => {
  const defaultProps = {};

  const renderComponent = overrides =>
    render(<Profile {...defaultProps} {...overrides} />);

  beforeEach(() => {
    const useUserRegister = td.replace(MutationHooks, 'useUserRegister');

    td.when(useUserRegister()).thenReturn([
      jest.fn().mockName('userRegister'),
      { loading: false },
    ]);
  });

  test('renders properly', () => {
    const { container } = renderComponent();

    expect(container).toMatchSnapshot();
  });
});
