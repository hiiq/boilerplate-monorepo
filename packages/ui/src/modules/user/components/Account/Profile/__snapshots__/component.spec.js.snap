// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile/> renders properly 1`] = `
<div>
  <x-breadcrumbs>
    <x-breadcrumb
      text="t(account)"
      to="/account"
    />
    <x-breadcrumb
      isend="true"
      text="t(profile)"
      to="/account/profile"
    />
  </x-breadcrumbs>
  <x-header
    title="t(profile)"
  />
  <x-body>
    <x-form />
  </x-body>
</div>
`;
